### PR TITLE
Add redis cache (ported to new codebase)

### PR DESCRIPTION
**Pull request by bmanojlovic**
I have refactored code from (#116) still did not test all features but it should work if old code was working.

Will test it a bit more from real system and report back in this PR

### CODE REVIEW

File: include/mapcache.h [hunk 1]
```diff
@@ -51,6 +51,10 @@
 #include <apr_time.h>
 
 
+#ifdef USE_REDIS
+#include <hiredis/hiredis.h>
+#endif
+
```
Comment (jbo-ads) [inline include/mapcache.h:57]: For consistency, I would exclude these lines from mapcache.h. On other caches, this optional include is located in the corresponding cache_{}.c file.

Comment (jbo-ads) [inline include/mapcache.h:57]: For consistency, I would exclude these lines from mapcache.h. On other caches, this optional include is located in the corresponding cache_{}.c file.

File: lib/cache_redis.c [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+  return pc;
+
+  // redisContext *conn;
+  // conn = redisConnect(cache->host, cache->port);
+  // if (!conn || conn->err) {
+  //   ctx->set_error(ctx,500, "redis: failed to connect to server %s:%d", cache->host, cache->port);
+  //   return NULL;
+  // }
+  // return conn;
```
Comment (jbo-ads) [inline lib/cache_redis.c:100]: Better to remove commented out code.

**Review by jbo-ads — COMMENTED** [review]
Thank you for submitting this new feature!

Here are my few comments on your code. I didn't check the actual logic of redis management. I hope your tests will detect any misbehaviours.

File: lib/cache_redis.c (hunk 2 above)
Comment (bmanojlovic) [reply lib/cache_redis.c]: suggested changes performed

**Comment by bmanojlovic** [comment]
> Thank you for submitting this new feature!
regarding tests performed, it works fine for me from mapcache_seed and from running as apache module

**Pull request merged**